File: php/features.php
<?php
declare(strict_types=1);

// BaconIpsum SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class BaconIpsumFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BaconIpsumBaseFeature();
            case "test":
                return new BaconIpsumTestFeature();
            default:
                return new BaconIpsumBaseFeature();
        }
    }
}

File: py/features.py
# BaconIpsum SDK feature factory

from feature.base_feature import BaconIpsumBaseFeature
from feature.test_feature import BaconIpsumTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BaconIpsumBaseFeature(),
        "test": lambda: BaconIpsumTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
